src/wallet/wallet2.cpp

```diff
@@ -8529,7 +8553,7 @@ std::vector<size_t> wallet2::pick_preferred_rct_inputs(uint64_t needed_money, ui
   for (size_t i = 0; i < m_transfers.size(); ++i)
   {
     const transfer_details& td = m_transfers[i];
-    if (!td.m_spent && !td.m_frozen && td.is_rct() && td.amount() >= needed_money && is_transfer_unlocked(td) && td.m_subaddr_index.major == subaddr_account && subaddr_indices.count(td.m_subaddr_index.minor) == 1)
+    if (!is_spent(td, false) && !td.m_frozen && td.is_rct() && td.amount() >= needed_money && is_transfer_unlocked(td) && td.m_subaddr_index.major == subaddr_account && subaddr_indices.count(td.m_subaddr_index.minor) == 1)
```

Comment:
`td.m_spent` is equivalent to `is_spent(td, false)`. Some instances of `td.m_spent` are replaced like this while others are left unchanged, which feels confusing/arbitrary to me. What's the point of these changes?

Reply:
Since the patch introduces a difference in semantics between "spent on the chain" and "spent on the chain or pending txes we just sent but are not yet mined", it seemed best to replace m_spent checks with is_spent to explicitely state which semantics we want, otherwise it seems dangerous. The ones that were not replaced are (1) the ones dealing with the m_spent value directly (ie, process_new_transaction, rescanning key images) and (2) the light wallet code, which I have no idea what its semantics are. I've just looked at the list and it seems fine to me. (1) can seen subjective though, so if you can point to specific instances I can check again.